### PR TITLE
Dance share and use new api 0.0.6

### PR DESCRIPTION
**Pull request by cpirich**
* Builds upon https://github.com/code-dot-org/code-dot-org/pull/25582
* Pulls in new Dance-Party 0.0.6 (see https://github.com/code-dot-org/dance-party/pull/42)
* Fixes share mode so we "autorun" and don't show the big "How it works" button
* Fixes running before p5's setup has completed (this would cause errors) (you can repro this in current builds by reloading without cache and pressing Run quickly - boom JS exception!)
* Also, adapts our normal Run flow to wait for the song metadata to load (this is an improvement because before we'd just fail the Run click if the metadata wasn't loaded. You'd have to click the button again until it worked)

### CODE REVIEW

File: apps/src/dance/Dance.js [hunk 1]
```diff
@@ -451,24 +455,41 @@ Dance.prototype.shouldShowSharing = function () {
...
+
 /**
  * This is called while this.p5 is in the setup phase.
  */
 Dance.prototype.onP5Setup = function () {
+  this.preloadComplete = true;
   this.nativeAPI.setup();
+  this.p5setupPromiseResolve();
+  if (this.share) {
+    setTimeout(this.studioApp_.runButtonClick, 0);
```
Comment (epeach) [inline apps/src/dance/Dance.js:491]: Is this `setTimeout` necessary?

File: apps/src/dance/Dance.js [hunk 2]
```diff
@@ -451,24 +455,41 @@ Dance.prototype.shouldShowSharing = function () {
...
   const spriteConfig = new Function('World', this.level.customHelperLibrary);
   this.nativeAPI.init(spriteConfig);
   this.nativeAPI.preload();
 };
 
+Dance.prototype.updateSongMetadata = function (id) {
+  this.songMetadataPromise = this.loadSongMetadata(id);
+};
+
+Dance.prototype.playSong = function (file, callback) {
```
Comment (epeach) [inline apps/src/dance/Dance.js:473]: `playSong` is not used in this set of changes.

Comment (cpirich) [reply apps/src/dance/Dance.js:473]: good point - I'll remove it.

File: apps/src/dance/Dance.js (hunk 1 above)
Comment (cpirich) [reply apps/src/dance/Dance.js:491]: probably not, I was imitating some gamelab code, but I can just do it directly.

**Comment by cpirich** [comment]
@epeach The share experience probably needs some UI tests. I don't think it is has any right now for any of the apps, but I'm not certain. I'm working on fixing some other significant UI issues with the mobile share experience that impacts all apps (not just dance party). I'll probably ask for some help with getting UI test coverage once I'm done.

**Pull request merged**